File: website/src/types/smtp.ts
export interface SMTPTest {
  request_id: string
  config_etag: string
  status: 'pending' | 'accepted' | 'failed' | 'unknown'
  code?: string
  duration_ms: number
  stages: { name: string; status: 'passed' | 'failed'; duration_ms: number; code?: string }[]
  started_at: string
  completed_at: string | null
}
export interface SMTPSettings {
  enabled: boolean
  host: string
  port: number
  security: 'STARTTLS' | 'SSL_TLS' | 'NONE'
  auth_configured: boolean
  sender_name: string
  sender_email: string
  reply_to: string
  etag: string
  updated_at: string
  last_test: SMTPTest | null
}
export interface SMTPInput {
  enabled: boolean
  host: string
  port: number
  security: SMTPSettings['security']
  auth: { action: 'keep' | 'replace' | 'remove'; username?: string; password?: string }
  etag: string
}
export interface SMTPSenderInput {
  sender_name: string
  sender_email: string
  reply_to: string
  etag: string
}
export interface SMTPTestInput {
  recipient: string
  request_id: string
  etag: string
}

File: internal/routex/entity/smtp.go
package entity

import "time"

type SMTPSetting struct {
	ID                int    `gorm:"primaryKey;autoIncrement:false"`
	Enabled           bool   `gorm:"not null"`
	Host              string `gorm:"size:253;not null"`
	Port              int    `gorm:"not null"`
	Security          string `gorm:"size:12;not null"`
	SecretGeneration  string `gorm:"size:30;not null" json:"-"`
	AuthCiphertext    string `gorm:"type:text;not null" json:"-"`
	SenderName        string `gorm:"size:100;not null"`
	SenderEmail       string `gorm:"size:254;not null"`
	ReplyTo           string `gorm:"size:254;not null"`
	ETag              string `gorm:"size:30;not null"`
	LastTestStartedAt *time.Time
	UpdatedAt         time.Time
}

type SMTPTest struct {
	ID              string    `gorm:"primaryKey;size:30"`
	RequestID       string    `gorm:"size:80;not null"`
	RequestHash     string    `gorm:"size:64;not null;uniqueIndex:idx_smtp_tests_request" json:"-"`
	ActorID         string    `gorm:"size:30;not null"`
	ConfigETag      string    `gorm:"size:30;not null"`
	RecipientDigest string    `gorm:"size:64;not null" json:"-"`
	Status          string    `gorm:"size:12;not null"`
	ResultJSON      string    `gorm:"type:text;not null"`
	StartedAt       time.Time `gorm:"not null;index:idx_smtp_tests_started"`
	CompletedAt     *time.Time
}

func (SMTPSetting) TableName() string { return "smtp_settings" }
func (SMTPTest) TableName() string    { return "smtp_tests" }
